File: 2015/Day10/main.lua
local function look_and_say(input)
    local result = {}
    local count = 1

    for i = 2, #input + 1 do
        if i <= #input and input:sub(i, i) == input:sub(i - 1, i - 1) then
            count = count + 1
        else
            table.insert(result, tostring(count) .. input:sub(i - 1, i - 1))
            count = 1
        end
    end

    return table.concat(result)
end

local input = '3113322113'

for i = 1, 40 do
    input = look_and_say(input)
end
print('Part 1:', #input)

for i = 1, 10 do
    input = look_and_say(input)
end
print('Part 2:', #input)

File: 2015/Day10/main.dart
String lookAndSay(String input) {
  var result = StringBuffer();
  var count = 1;

  for (var i = 1; i <= input.length; ++i) {
    if (i < input.length && input[i] == input[i - 1]) {
      count++;
    } else {
      result.write('$count${input[i - 1]}');
      count = 1;
    }
  }

  return result.toString();
}

void main() {
  var input = '3113322113';

  for (var i = 0; i < 40; i++) {
    input = lookAndSay(input);
  }

  print('Part 1: ${input.length}');

  for (var i = 0; i < 10; i++) {
    input = lookAndSay(input);
  }

  print('Part 2: ${input.length}');
}
